src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+            comp.VerifyDiagnostics(
+                // (8,28): error CS8803: 'DerivedWithInit.Property' must match by init-only of overridden member 'Base.Property'
+                //     public override string Property { get; init; } // 1, 2
+                Diagnostic(ErrorCode.ERR_CantChangeInitOnlyOnOverride, "Property").WithArguments("DerivedWithInit.Property", "Base.Property").WithLocation(8, 28),
+                // (8,39): error CS0545: 'DerivedWithInit.Property.get': cannot override because 'Base.Property' does not have an overridable get accessor
+                //     public override string Property { get; init; } // 1, 2
+                Diagnostic(ErrorCode.ERR_NoGetToOverride, "get").WithArguments("DerivedWithInit.Property.get", "Base.Property").WithLocation(8, 39),
+                // (12,39): error CS0545: 'DerivedWithoutInit.Property.get': cannot override because 'Base.Property' does not have an overridable get accessor
+                //     public override string Property { get; set; } // 3
+                Diagnostic(ErrorCode.ERR_NoGetToOverride, "get").WithArguments("DerivedWithoutInit.Property.get", "Base.Property").WithLocation(12, 39),
+                // (16,28): error CS8803: 'DerivedWithInitSetterOnly.Property' must match by init-only of overridden member 'Base.Property'
+                //     public override string Property { init { } } // 4
+                Diagnostic(ErrorCode.ERR_CantChangeInitOnlyOnOverride, "Property").WithArguments("DerivedWithInitSetterOnly.Property", "Base.Property").WithLocation(16, 28),
+                // (20,39): error CS0545: 'DerivedWithoutInitGetterOnly.Property.get': cannot override because 'Base.Property' does not have an overridable get accessor
+                //     public override string Property { get => null; } // 5
+                Diagnostic(ErrorCode.ERR_NoGetToOverride, "get").WithArguments("DerivedWithoutInitGetterOnly.Property.get", "Base.Property").WithLocation(20, 39)
+                );
+        }
+
+        [Fact]
+        public void ImplementingInitOnlyProperty()
+        {
+            string source = @"
+public interface I
+{
+    string Property { get; init; }
+}
+public class DerivedWithInit : I
+{
+    public string Property { get; init; }
+}
+public class DerivedWithoutInit : I // 1
+{
+    public string Property { get; set; }
+}
+public class DerivedWithInitSetterOnly : I // 2
+{
+    public string Property { init { } }
+}
+public class DerivedWithoutInitGetterOnly : I // 3
+{
+    public string Property { get => null; }
+}
+";
+
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (10,35): error CS8804: 'DerivedWithoutInit' does not implement interface member 'I.Property.set'. 'DerivedWithoutInit.Property.set' cannot implement 'I.Property.set' because it does not match by init-only.
```

Comment:
>set [](start = 180, length = 3)

Should we say "init" here and if so do we really need  to say "because it does not match by init-only"?

Reply:
Maybe.
For the moment, we say "set". See PROTOTYPE comments on symbol display and public API. I do not want to expand the scope of this PR. 
When that changes, we can review the diagnostic message. 

---
In reply to: [415205267](https://github.com/dotnet/roslyn/pull/43275#discussion_r415205267) [](ancestors = 415205267)

Reply:
> SymbolDisplay didn't affect this. I plan to keep the name of the symbol and the diagnostic as-is.

I'd like to make sure I understand what you mean. Are you saying that you are not planning adjust symbol display to say "I.Property.init" not just in this PR, but ever?

---
In reply to: [415330627](https://github.com/dotnet/roslyn/pull/43275#discussion_r415330627) [](ancestors = 415330627,415205267)

Reply:
Correct

---
In reply to: [415985052](https://github.com/dotnet/roslyn/pull/43275#discussion_r415985052) [](ancestors = 415985052,415330627,415205267)

Reply:
> Correct

But you are making changes in ```SymbolDisplayVisitor.VisitProperty``` to actually display ```init``` instead of ```set```. And here you are saying that in other perhaps in other part of symbol display, you are not planning to make any changes. This feels like the approach is somewhat inconsistent. Don't you?

---
In reply to: [415994350](https://github.com/dotnet/roslyn/pull/43275#discussion_r415994350) [](ancestors = 415994350,415985052,415330627,415205267)

Reply:
I think the design of having an accessor with a different name but that is still a `set` accessor puts us in this bind. The question is where we draw the line (where do we stop calling it `init`?).
The reason I changed symbol display is because it prints something like syntax: `int Property { get; set/init; }`. But then the accessor is still a `set`, so `Property.set` seems okay... Sticking with `set` throughout (except for syntax) would be simpler?

I honestly don't feel strongly about that. Wherever we draw the line, there will be a discontinuity :-P

---
In reply to: [416745129](https://github.com/dotnet/roslyn/pull/43275#discussion_r416745129) [](ancestors = 416745129,415994350,415985052,415330627,415205267)